### PR TITLE
Change Creator from plan to product

### PR DESCRIPTION
**Pull request by robertsreberski**
Related to #

## Proposed Changes

*

## Testing Instructions

*

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/calypso-products/src/constants/jetpack.ts [hunk 1]
```diff
@@ -800,6 +803,12 @@ export const JETPACK_PRODUCT_RECCOMENDATION_MAP: Record< string, string[] > = {
 		JETPACK_TAG_FOR_ONLINE_FORUMS,
 		JETPACK_TAG_FOR_WOOCOMMERCE_STORES,
 	],
+	[ PRODUCT_JETPACK_CREATOR_BI_YEARLY ]: [
```
Comment (elliottprogrammer) [inline packages/calypso-products/src/constants/jetpack.ts]: Looks like this needs to be: `PRODUCT_JETPACK_CREATOR_BI_YEARLY`

```suggestion
	[ PRODUCT_JETPACK_CREATOR_YEARLY ]: [
```

**Review by elliottprogrammer — APPROVED** [review]
I just see one tiny constant that needs fixed (below in the inline comments). Otherwise, this tests well, and the code looks all good!
LGTM! 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9647342

Some locales (Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @robertsreberski, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Craft stunning content, boost your subscriber base, and monetize your online presence.`
- `Quickly create content that stands out`
- `Grow your subscribers with simple subscribe forms`
- `Create content for paid subscribers`
- `Sell access to premium content`
- `Easily accept tips and donations`
- `Creator`
- `You have tools to create, grow, and monetize your audience.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9647342).

**Pull request merged**